File: src/javaStuff/oopStuff/abstractAndInterface/Cat.java
package javaStuff.oopStuff.abstractAndInterface;

public class Cat extends Animal implements AnimalStuff {
    // we get nickname from AnimalStuff, but we can not change the name
    String catNickname = nickname;
    @Override
    public void eat() {
        System.out.println("the cat eating now");
    }
    @Override
    public void Poop() {
        super.Poop();
    }
    @Override
    public String makeNoise() {
        return "Meow";
    }

    @Override
    public void jump() {
        System.out.println("cat jump highly");
    }

    @Override
    public String swim() {
        return "cat can swim but do not like swim";
    }
}

File: src/kotlinStuff/abstractAndInterface/Cat.kt
package kotlinStuff.abstractAndInterface

import javaStuff.abstractAndInterface.AnimalStuff

class Cat: Animal(), AnimalStuff {
    // we get nickname from AnimalStuff, but we can not change the name
    var catNickname = AnimalStuff.nickname

    override fun jump() { println("this cat can jump") }

    override fun swim(): String { return "can swim" }

    override fun makeNoise() { println("it's make a lot of noises") }

    override fun eye(): String { return "good eye" }

    override fun poop() {
        super.poop()
        println("SecondPoopPrint")
    }
}